src/qt/window.cpp

```diff
@@ -274,27 +274,33 @@ static const char WINDOW_POINTER_PROPERTY_NAME[] = "wxWindowPointer";
 }
 
 /* static */
-void wxWindowQt::QtSendSetCursorEvent(wxWindowQt* win, wxPoint posScreen)
+void wxWindowQt::QtSendSetCursorEvent(wxWindowQt* win, const wxPoint& posClient)
 {
-    wxWindowQt* w = win;
-    for ( ;; )
+    const wxRect rect(win->GetClientAreaOrigin(), win->GetClientSize());
+
+    if ( rect.Contains(posClient) )
     {
-        const wxPoint posClient = w->ScreenToClient(posScreen);
-        wxSetCursorEvent event(posClient.x, posClient.y);
-        event.SetEventObject(w);
+        wxSetCursorEvent event( posClient.x , posClient.y );
+        event.SetId(win->GetId());
+        event.SetEventObject(win);
 
-        const bool processedEvtSetCursor = w->ProcessWindowEvent(event);
+        bool processedEvtSetCursor = win->HandleWindowEvent(event);
```

Comment:
Very minor, but looks like `const` could remain:

```suggestion
        const bool processedEvtSetCursor = win->HandleWindowEvent(event);
```